src/movies.js

```diff
@@ -1,25 +1,69 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+function getAllDirectors(moviesArray) {
+  const directors = moviesArray.map(movie => movie.director)
+return directors
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+  return moviesArray.filter(movie => movie.genre.includes("Drama") && movie.director === "Steven Spielberg").length
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+function scoresAverage(moviesArray) {
+  if (moviesArray.length === 0){
+    return 0
+  }
+  const scores = moviesArray.map(movie => movie.score)
+  const sum = scores.reduce((acc, curr) => {
+    return acc + curr
+  })
+  const mean = sum / scores.length
+  return parseFloat(mean.toFixed(2))
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore(moviesArray) {}
+function dramaMoviesScore(moviesArray) {
+  const dramas = moviesArray.filter(movie => movie.genre.includes("Drama"))
+  if (dramas.length === 0){
+    return 0
+  }
+  return scoresAverage(dramas)
+}
 
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-function orderByYear(moviesArray) {}
+function orderByYear(moviesArray) {
+  const arrayToSort = [...moviesArray]
+  arrayToSort.sort((movie1, movie2) => movie1.title - movie2.title)
+  arrayToSort.sort((movie1, movie2) => movie1.year - movie2.year)
```

Comment:
You probably want to sort the title and the year at the same time

Just an example: 

```
function orderByYear(moviesArray) {
  const arrayToSort = [...moviesArray]
  // arrayToSort.sort((movie1, movie2) => movie1.title - movie2.title)
  arrayToSort.sort((movie1, movie2) => {
    if (movie1.year === movie2.year){
      let title1 = movie1.title.toUpperCase();
      let title2 = movie2.title.toUpperCase();
      if (title1 < title2) return -1;
      if (title1 > title2) return 1;
    }
    return movie1.year - movie2.year
  })
  return arrayToSort
}
```